### PR TITLE
Implement mkfifo command

### PR DESCRIPTION
**Pull request by bkochendorfer**
Add new mkfifo command and tests. I'm on the fence on if this should be in `pkg` or just all in `cmd` so let me know what you think. The default mode is cribbed from the existing mkfifo command, if there is a simpler way to express that I'd love to hear about it.

### CODE REVIEW

File: pkg/mkfifo/mkfifo.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+package mkfifo
+
+import "syscall"
+
+type Mkfifo struct {
+	Paths []string
+	Mode  uint32
+}
+
+func (m *Mkfifo) Exec() error {
```
Comment (hugelgupf) [inline pkg/mkfifo/mkfifo.go]: This is simple enough that I don't think it warrants a package.

File: pkg/mkfifo/mkfifo.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	var err error
```
Comment (hugelgupf) [inline pkg/mkfifo/mkfifo.go]: Unused variable -- you could ust

```go
for _, path := range m.Paths {
   if err := ...; err != nil {
     return err
   }
}
return nil
```

File: pkg/mkfifo/mkfifo.go (hunk 1 above)
Comment (GanShun) [reply pkg/mkfifo/mkfifo.go]: I agree with Chris here, I think its simple enough that if you need it from somewhere else we could just reimplement it.

Comment (rminnich) [reply pkg/mkfifo/mkfifo.go]: or, OTOH, we could add FIFO as one of the things we support in pkg/u-root/util/root.go, along with directories, files, etc.

File: cmds/mkfifo/mkfifo.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import (
+	"flag"
+	"log"
+	"syscall"
+
+	"github.com/u-root/u-root/pkg/mkfifo"
+)
+
+var (
+	defaultMode = syscall.S_IRUSR | syscall.S_IWUSR | syscall.S_IRGRP | syscall.S_IWGRP | syscall.S_IROTH | syscall.S_IWOTH
```
Comment (rminnich) [inline cmds/mkfifo/mkfifo.go]: SHouldn't this be a const?

**Comment by rjoleary** [comment]
How does this differ from mknod with the p (pipe) option?

https://github.com/u-root/u-root/blob/master/cmds/mknod/mknod_linux.go#L63

**Comment by rminnich** [comment]
we need to up our game on documentation, clearly. I think the table with mkfifo ought to be changed to point to mknod?

**Comment by hugelgupf** [comment]
I feel like mkfifo is more common than mknod. And I've seen it in plenty of bash scripts. It's small enough that we might as well have both?

**Comment by rminnich** [comment]
I'm fine with adding it if that seems to simplify lives.

**Comment by rjoleary** [comment]
Sounds good! I'm fine with having both.

The table contains mkfifo because petitboot contains mkfifo. At one point, we wanted it be easy to convert petitpoot scripts to u-root, so all the petitboot commands went into the table.

One additional difference I see is in the default mode. u-root's mknod uses 660 and this new mkfifo uses 666. It would be worth making them both uniform. Personally, I'd prefer 660 for both.

**Comment by rjoleary** [comment]
Also, I remember Ron once wrote some guidelines on introducing new commands, comparing them to linear dependent and independent vectors. I believe it was in the mailing list. @rminnich Do you still have the link?

File: cmds/mkfifo/mkfifo.go (hunk 3 above)
Comment (rjoleary) [reply cmds/mkfifo/mkfifo.go]: Have you consider using 660 as the default mode? It would be tad bit more secure.

**Comment by rminnich** [comment]
I just found and reposted that comment.

File: cmds/mkfifo/mkfifo.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+func main() {
+	flag.Parse()
+
+	if flag.NArg() < 1 {
+		log.Fatal("please provide a path, or multiple, to create a fifo")
+	}
+
+	for _, path := range flag.Args() {
+		if err := syscall.Mkfifo(path, uint32(*mode)); err != nil {
```
Comment (hugelgupf) [inline cmds/mkfifo/mkfifo.go]: super nit: let's use either syscall or unix, but not both? (i'd go for unix).

**Comment by bkochendorfer** [comment]
Hey all, looks like I don't have the ability to merge. If all looks well go ahead and merge.

File: cmds/mkfifo/mkfifo_test.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+		for _, path := range tt.flags {
+			testFile, err := os.Stat(path)
+
+			if err != nil {
+				t.Errorf("Unable to stat file %s", path)
+			}
+
+			mode := testFile.Mode()
+			if (mode & os.ModeNamedPipe).String() != "p---------" {
```
Comment (hugelgupf) [inline cmds/mkfifo/mkfifo_test.go]: `if mode & os.ModeNamedPipe != 0`

Comment (rminnich) [reply cmds/mkfifo/mkfifo_test.go]: The second one. Some of the types have multiple bits set so we might as well code that way.

Comment (bkochendorfer) [reply cmds/mkfifo/mkfifo_test.go]: Great feedback and thanks for the link!

**Pull request merged**